File: cpp/Motherboard.cpp
// Motherboard.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"
using namespace std;

class Motherboard : public Komponen {
private:
    string chipset;
    string formFactor;
    int maxRamSlots;
    
public:
    Motherboard() {}
    
    Motherboard(string chipset, string formFactor, int maxRamSlots, string merk, string nama) : Komponen(merk, nama) {
        this->chipset = chipset;
        this->formFactor = formFactor;
        this->maxRamSlots = maxRamSlots;
    }
    
    void setChipset(string chipset) {
        this->chipset = chipset;
    }
    
    void setFormFactor(string formFactor) {
        this->formFactor = formFactor;
    }
    
    void setMaxRamSlots(int maxRamSlots) {
        this->maxRamSlots = maxRamSlots;
    }
    
    string getChipset() {
        return this->chipset;
    }
    
    string getFormFactor() {
        return this->formFactor;
    }
    
    int getMaxRamSlots() {
        return this->maxRamSlots;
    }
    
    ~Motherboard() {}
};

File: python/Motherboard.py
from Komponen import Komponen
class Motherboard(Komponen) :
    def __init__(self, chipset, formFactor, maxRamSlots, merk, nama):
        super().__init__(merk, nama)
        
        self.__chipset = chipset
        self.__formFactor = formFactor
        self.__maxRamSlots = maxRamSlots
        
    def get_chipset(self):
        return self.__chipset
    def get_formFactor(self):
        return self.__formFactor
    def get_maxRamSlots(self):
        return self.__maxRamSlots
    
    # Setter methods
    def set_chipset(self, chipset):
        self.__chipset = chipset
    def set_formFactor(self, formFactor):
        self.__formFactor = formFactor
    def set_maxRamSlots(self, maxRamSlots):
        self.__formFactor = maxRamSlots
        